schema/oneof_test.go

```diff
@@ -140,68 +127,94 @@ var oneOfTestCMappedSchema = schema.NewStructMappedObjectSchema[oneOfTestObjectC
 	oneOfTestObjectCProperties,
 )
 
-var oneOfTestInlineObjectBProperties = map[string]*schema.PropertySchema{
-	"message": schema.NewPropertySchema(
-		schema.NewStringSchema(nil, nil, nil),
-		nil,
-		true,
-		nil,
-		nil,
-		nil,
-		nil,
-		nil,
-	),
-	"choice": schema.NewPropertySchema(
-		schema.NewStringSchema(nil, nil, nil),
-		nil,
-		true,
-		nil,
-		nil,
-		nil,
-		nil,
-		nil,
-	),
-}
+// Test_OneOf_ConstructorBypass tests the behavior of a OneOf object created
+// by the Scope Scope Schema through unserialization of data without using a
+// New* constructor function, like NewOneOfStringSchema or NewOneOfIntSchema,
+// behaves as one would expect from a OneOf object created from a constructor.
+func Test_OneOf_ConstructorBypass(t *testing.T) { //nolint:funlen
+	discriminator_field := "_type"
+	input_schema := map[string]any{
+		"root": "InputParams",
+		"objects": map[string]any{
+			"InputParams": map[string]any{
+				"id": "InputParams",
+				"properties": map[string]any{
+					"name": map[string]any{
+						"required": true,
+						"type": map[string]any{
+							"discriminator_field_name": discriminator_field,
+							"type_id":                  "one_of_string",
+							"types": map[string]any{
+								"fullname": map[string]any{
+									"id":      "FullName",
+									"type_id": "ref",
+								},
+								"nick": map[string]any{
+									"id":      "Nickname",
+									"type_id": "ref",
+								},
+							},
+						},
+					},
+				},
+			},
+			"FullName": map[string]any{
+				"id": "FullName",
+				"properties": map[string]any{
+					"first_name": map[string]any{
+						"required": true,
+						"type": map[string]any{
+							"type_id": "string",
+						},
+					},
+					"last_name": map[string]any{
+						"required": true,
+						"type": map[string]any{
+							"type_id": "string",
+						},
+					},
+				},
+			},
+			"Nickname": map[string]any{
+				"id": "Nickname",
+				"properties": map[string]any{
+					"nick": map[string]any{
+						"required": true,
+						"type": map[string]any{
+							"type_id": "string",
+						},
+					},
+				},
+			},
+		},
+	}
+	var input_data_fullname any = map[string]any{
+		"name": map[string]any{
+			discriminator_field: "fullname",
+			"first_name":        "Arca",
+			"last_name":         "Lot",
+		},
+	}
+
+	scopeAny := assert.NoErrorR[any](t)(schema.DescribeScope().Unserialize(input_schema))
+	scopeSchemaTyped := scopeAny.(*schema.ScopeSchema)
+	scopeSchemaTyped.ApplyScope(scopeSchemaTyped)
+	assert.NoError(t, scopeSchemaTyped.Validate(input_data_fullname))
+	unserialized := assert.NoErrorR[any](t)(scopeSchemaTyped.Unserialize(input_data_fullname))
+	serialized := assert.NoErrorR[any](t)(scopeSchemaTyped.Serialize(unserialized))
+	unserialized2 := assert.NoErrorR[any](t)(scopeSchemaTyped.Unserialize(serialized))
+	assert.Equals(t, unserialized2, unserialized)
+
+	var input_invalid_discriminator_value any = map[string]any{
+		"name": map[string]any{
+			discriminator_field: 1,
```

Comment:
I'm sure that this is a good test, but can a discriminator field ever have a numerical type?  Shouldn't this (or an additional scenario) be using a string whose value isn't in the schema?